internal/encoding/asn1/asn1.go

```diff
@@ -0,0 +1,235 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package asn1 decodes BER-encoded ASN.1 data structures and encodes in DER.
+// Note: DER is a subset of BER.
+// Reference: http://luca.ntop.org/Teaching/Appunti/asn1.html
+package asn1
+
+import (
+	"bytes"
+	"encoding/asn1"
+	"math"
+)
+
+// Common errors
+var (
+	ErrBytesAtTheEnd              = asn1.StructuralError{Msg: "invalid bytes at the end of the BER data"}
```

Comment:
It is commonly called `asn1.SyntaxError{Msg: "trailing data"}`.

See https://cs.opensource.google/search?q=%22trailing%20data%22&sq=&ss=go%2Fgo

Reply:
Updated.